### ISSUE TITLE
AddOutgoingHeader does not work inside filter

### ISSUE THREAD

**Issue by miyconst** [Starcounter/Starcounter#2961]
This header never reach client side.

```cs
Handle.AddFilterToMiddleware(request => {
	Handle.AddOutgoingHeader("SignInApp", DateTime.Now.ToString());

	return null;
});
```

**Comment by warpech** [comment]
Mind that `AddFilterToMiddleware` is a request filter so this might be by design. Is this needed? 

@alemoi can you elaborate how this works? If this is by design, could you provide a meaningful exception when `AddOutgoingHeader` is used in wrong context?

**Comment by alemoi** [comment]
Actually I just fixed the issue via b2f8c1637131fe9f0cba903debfffc42ad3cd90b
Available in this nightly.
I have rewritten the code regarding how outgoing response fields are processed, now, for example, if one adds a header, then all responses, including static files, will contain that header.
It was working wrong before (not covering all cases), and setting outgoing headers in `AddFilterToMiddleware` is totally fine.

**Comment by miyconst** [comment]
Confirm to be fixed.
